workflow.swift

```diff
@@ -56,14 +69,15 @@ wait(ready) {
         printf("swift: The launched application did not succeed.");
     }
 
-    program2 = "stage_write/stage_write";
-    arguments2 = split("heat.bp staged.bp %s \"\" MPI \"\"" % rmethod , " ");
-    printf("size: %i", size(arguments2));
-    printf("swift: launching: %s", program2);
-    exit_code2 = @par=swproc launch(program2, arguments2);
-    printf("swift: received exit code: %d", exit_code2);
-    if (exit_code2 != 0)
+    if(rmethod == "BP")
     {
-        printf("swift: The launched application did not succeed.");
+        wait (exit_code1)
```

Comment:
For the file-based method, it is safe for `stag_write` to wait until `heat_transfer` finishes. Since a same file is shared between two programs, there is a possibility that `stage_write` accesses uncompleted data and thus goes crash. For other staging methods (FLEXPATH or DATASPACES), the order doesn't matter.